### PR TITLE
Droth 909 tr to oth speed limits

### PR DESCRIPTION
**Pull request by marquesfm**

### CODE REVIEW

**Comment by leirasf** [comment]
Ok

File: UI/src/view/SpeedLimitForm.js [hunk 1]
```diff
@@ -129,6 +131,17 @@
...
 
+  function validateAdministrativeClass(selectedSpeedLimit){
+    var selectedSpeedLimits = _.filter(selectedSpeedLimit.get(), function (selected) {
+      return editConstrains(selected);
+    });
+    return !_.isEmpty(selectedSpeedLimits);
+  }
+
+  var editConstrains = function(selectedAsset) {
+    return selectedAsset.administrativeClass === "State";
```
Comment (joanacoelho) [inline UI/src/view/SpeedLimitForm.js]: To be coherent use ' instead of "

`'State'`

File: UI/src/view/SpeedLimitLayer.js [hunk 2]
```diff
@@ -548,6 +557,10 @@ window.SpeedLimitLayer = function(params) {
       me.refreshView();
   };
 
+  var editConstrains = function(selectedAsset) {
+    return selectedAsset.administrativeClass === "State";
```
Comment (joanacoelho) [inline UI/src/view/SpeedLimitLayer.js]: Same as above

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala [hunk 3]
```diff
@@ -689,4 +689,41 @@ class EuropeanRoadTierekisteriImporter extends LinearAssetTierekisteriImporterOp
...
+  override def assetName = "speedlimit"
+  override type TierekisteriClientType = TierekisteriSpeedLimitAssetClient
+  override def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+  override def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+
+  override val tierekisteriClient = new TierekisteriSpeedLimitAssetClient(getProperty("digiroad2.tierekisteriRestApiEndPoint"),
+    getProperty("digiroad2.tierekisteri.enabled").toBoolean,
+    HttpClientBuilder.create().build())
+
+  private def getSideCode(raSideCode: SideCode, roadSide: RoadSide): SideCode = {
```
Comment (joanacoelho) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala]: This method already exists on `SpeedLimitsTierekisteriImporter`

**Comment by joanacoelho** [comment]
Otherwise ok

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala (hunk 3 above)
Comment (oskarmalmstrom) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala]: But it is private. Or am i missing something?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/TierekisteriClient.scala [hunk 4]
```diff
@@ -955,6 +958,34 @@ class TierekisteriEuropeanRoadAssetClient(trEndPoint: String, trEnable: Boolean,
...
 
+class TierekisteriSpeedLimitAssetClient(trEndPoint: String, trEnable: Boolean, httpClient: CloseableHttpClient) extends TierekisteriAssetDataClient {
+  override def tierekisteriRestApiEndPoint: String = trEndPoint
+  override def tierekisteriEnabled: Boolean = trEnable
+  override def client: CloseableHttpClient = httpClient
+  type TierekisteriType = TierekisteriSpeedLimitData
+
+  override val trAssetType = "tl168"
+  private val trSpeedLimitValue = "NOPRAJ"
+  private val trPUOLI = "PUOLI"
```
Comment (marquesfm) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/TierekisteriClient.scala]: switch trPUOLI to English

**Pull request merged**